File: two-pointers/valid_palindrome.c
// https://leetcode.com/problems/valid-palindrome

#include <ctype.h>
#include <string.h>

#define TRUE 1
#define FALSE 0

int isPalindrome(char* s)
{
	char *l = s;
	char *r = s + strlen(s);

	while (l < r) {
		while (!isalnum(*l) && l < r)
			l++;
		while (!isalnum(*r) && l < r)
			r--;
		if (tolower(*(l++)) != tolower(*(r--)))
			return FALSE;
	}
	return TRUE;
}

File: two-pointers/valid_palindrome.py
# https://leetcode.com/problems/valid-palindrome

class Solution:
    def isPalindrome(self, s):
        pb = 0
        pe = -1

        s = "".join(c for c in s if c.isalnum())
        while pb < len(s):
            if s[pb].lower() != s[pe].lower():
                return False
            pb += 1
            pe -= 1

        return True


assert Solution().isPalindrome("A man, a plan, a canal: Panama")
